.github/workflows/e2e-tests.yml

```diff
@@ -255,3 +245,36 @@ jobs:
 
       - name: "Test"
         run: ${{ matrix.script }}
+
+  result-cache-e2e-tests-bashunit:
+    name: "Result cache E2E tests"
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - script: |
+              cd e2e/result-cache-1
+              bashunit .
```

Comment:
do we still want a single isolated github action job per test?

Reply:
Yes, because I'm worried about polluting the state and making the tests worthless.

Reply:
It'd be great if we could just use the asserts outside of running with bashunit.

Reply:
> It'd be great if we could just use the asserts outside of running with bashunit.

do you mean we should leave this script sectino in the e2e-test.yml file as before but just replacing `grep` with `assert_contains`? should I not add `test.js` like files in the test folders?

Reply:
Yes that's exactly what I wanted to do, but I don't know if bashunit supports that.

Reply:
so something like

```diff
              cd e2e/trait-caching
              ../../bin/phpstan analyze --no-progress --level 8 --error-format raw data/
              patch -b data/TraitOne.php < TraitOne.patch
              OUTPUT=$(../../bin/phpstan analyze --no-progress --level 8 --error-format raw data/ || true)
              echo "$OUTPUT"
              [ $(echo "$OUTPUT" | wc -l) -eq 1 ]
-             grep 'Method TraitsCachingIssue\\TestClassUsingTrait::doBar() should return stdClass but returns Exception.' <<< "$OUTPUT"
+             bashunit assert_contains 'Method TraitsCachingIssue\\TestClassUsingTrait::doBar() should return stdClass but returns Exception.' "$OUTPUT"

```

..?

Reply:
lets see https://github.com/TypedDevs/bashunit/issues/257

Reply:
Would this be what you are looking for? https://github.com/TypedDevs/bashunit/pull/258